tests/e2e/assistant.spec.js

```diff
@@ -1,19 +1,208 @@
 import { test, expect } from './fixtures';
 
-// test('example test', async ({ page }) => {
-//   await page.goto('https://example.com');
-//   await expect(page.locator('body')).toHaveText('Changed by my-extension');
-// });
-
-test('Assstant page', async ({ page, extensionId }) => {
-  // Navigate to the assistant page
-  await page.goto(`chrome-extension://${extensionId}/popup.html#/app/assistant/`);
-  // Accept local storage usage
-  await page.getByText("Accept").click();
-  await expect(page.getByTestId("url-media-results")).not.toBeVisible();
-  await page.getByTestId("assistant-webpage-link").click();
-  await page.locator("[data-testid='assistant-url-selected-input'] input").fill("https://www.youtube.com/watch?v=WaaL75G0qu0");
-  await page.getByTestId("assistant-url-selected-analyse-btn").click();
-  await expect(page.getByTestId("url-media-results")).toBeVisible();
-
-});
+
+const MediaType = {
+  video: "video",
+  image: "image"
+};
+
+const MediaVideoStatus = {
+  iframe: 0,
+  video: 1,
+  noEmbed: 2,
+};
+
+const MediaServices = {
+  analysisVideo: "navbar_analysis_video",
```

Comment:
To avoid having hardcoded strings, I introduced a constants folder in which there is the tools file. Please use this as this will be used in the codebase moving forward. This will also make your life easier if some keys change in the future

Reply:
Can you provide me a link to this constants folder so I can use the existing files as an example?

Reply:
Yes, you can reuse the constants in `src/constants/tools.jsx`

i.e. you can grab the `titleKeyword` (first property) of the videoAnalysis object:

```
const videoAnalysis = new Tool(
  "navbar_analysis_video",
  "navbar_analysis_description",
  AnalysisSvgIcon,
  TOOLS_CATEGORIES.VIDEO,
  null,
  null,
  "analysis",
  TOOL_GROUPS.VERIFICATION,
  <Analysis />,
  <Footer type={FOOTER_TYPES.ITI} />,
);
```

(you will need to export the object)

https://github.com/AFP-Medialab/verification-plugin/blob/beta-master/src/constants/tools.jsx

Reply:
@Sallaa unfortunately there's no easy way to import .jsx into normal javascript for running the playwright e2e tests. Also even if it's possible the constants file has a lot of component imports which will introduce unnecessary dependencies into the test file.

I think it's ok to keep the this list in the test separate from the constants file and if the keys should change in the app (which shouldn't be often), it will be flagged up in the tests anyway.

Reply:
I'm looking into updating the AssistantRuleBook to use the new constants, I just wanted to check with you how I can access a specific tool. 

Since all the tools inside the `tools` object is stored an array, to get a specific tool I'd need to iterate through each one and search for the right one? Also since there's no unique ID assigned to each tool, the only thing I can use to find the correct one is the `titleKeyword` or `descriptionKeyword` field. Would it not be better to store the tools list as a dictionary instead?

So we can use it like this (from `AssistantRuleBook.jsx`):
```
export const ASSISTANT_ACTIONS = [
  {
    title: tools.videoAnalysis.titleKeyword,
    icon: tools.videoAnalysis.icon,
    linksAccepted: [KNOWN_LINKS.YOUTUBE, KNOWN_LINKS.FACEBOOK],
    cTypes: [CONTENT_TYPE.VIDEO],
    exceptions: [],
    useInputUrl: true,
    text: tools.videoAnalysis.descriptionKeyword,
    tsvPrefix: "api",
    path: tools.videoAnalysis.path,
  },
```

Or is the plan to add the properties of the objects in ASSISTANT_ACTIONS into the constants file itself?

Reply:
Ah, sorry, I thought you were using a JSX file -- feel free to leave it as it is. The goal is not to add bulk and complexity

It is fine for me if you export a specific tool constant from the tools so that you do not need to iterate through the array. I have exported some of the tools already.

Speaking about the array, yes, it makes more sense to use some kind of a dictionary instead, I'll update this when I'll have some more time to continue the refactoring. I think I have an instance where I need to search by `titleKeyword` but it's not ideal at all. 🥲

Are you updating the `AssistantRuleBook` in this PR? Just to know if I can merge this or if I should wait a little more

Reply:
This PR is ready to go, but waiting for some translation checking in https://github.com/AFP-Medialab/InVID-Translations/pull/90. We need French, Italian and German.  

I might as well wait to update the `AssistantRuleBook` when you've refactored the constants file.